packages/jwt/lib/src/jwt.dart

```diff
@@ -39,19 +30,23 @@ Future<Map<String, String>> _getPublicKeys(String url) async {
   final maxAgeRegExp = RegExp(r'max-age=(\d+)');
   final match = maxAgeRegExp.firstMatch(response.headers['cache-control']!);
   final maxAge = int.parse(match!.group(1)!);
-  final publicKeys = (json.decode(response.body) as Map<String, dynamic>)
-      .cast<String, String>();
 
-  _publicKeyStore = PublicKeyStore(
-    keys: publicKeys,
-    expiration: clock.now().add(Duration(seconds: maxAge)),
+  final publicKeyStore = PublicKeyStore.tryDeserialize(
+    json.decode(response.body) as Map<String, dynamic>,
   );
```

Comment:
I would probably have directly configured somewhere that Google is KVS vs microsoft JWK or something, rather than having this automatic fallback.  But this is also fine too, and not worth futzing over. :)